io_scene_godot/structures.py

```diff
@@ -93,10 +97,20 @@ def fix_paths(self, export_settings):
 
     def to_string(self):
         """Serializes the file ready to dump out to disk"""
+        ires = []
```

Comment:
I think if you always put `InternalResource()` into `self. internal_resources`, we would not need this change right?

Reply:
this change is helpful for others who are quickly jumping into this code base, adding some feature they need, and don't have much time to read how the exporter is designed.  By providing the option to pass a list or string, the exporter becomes easy to customize with minimal understanding of its internals.

Reply:
I understand your point, to make the code much flexible and context-free for new developers.  
However, I kind of disagree with this part of change, I also want the code to be manageable. For the InternalResource case, I think it is better just delegate all the serialization work to the InternalResource class. If you want insert a list of InternalResource objects, just add it to escn_file one by one.

Also the InternalResource is strictly corresponding to Resource in Godot, we are use it to detect multiple references to same Resource and avoid exporting one Resource more than once. Your change would break this functionality.

Reply:
my change breaks nothing.  Anyone who passes a list or string to that function, knows what they are doing, because they are adding some new generator, or inserting their own custom stuff, they understand and may prefer to manually manage the resource ids.